### PR TITLE
Document default CSP #371

### PR DESCRIPTION
**Pull request by rymsha**

### CODE REVIEW

**Comment by sigdestad** [comment]
A bit unclear this one: "page.defaultContentSecurityPolicy" - Does it apply to site engine, webapps, admin - or all? If this header only applies to site engine, I think it should be called "site.default.." And explicitly state it only relates to site engine.

What about admin, shouldn't there be a default security policy for admin apps as well?

**Comment by sigdestad** [comment]
Also, how does this relate to the "default headers" config that can be set for the web config?

**Comment by alansemenov** [comment]
> A bit unclear this one: "page.defaultContentSecurityPolicy" - Does it apply to site engine, webapps, admin - or all? If this header only applies to site engine, I think it should be called "site.default.." And explicitly state it only relates to site engine.

If we're going to ditch "`site`" in favour of "`cms`", maybe it should be called "`cms.defaultContentSecurityPolicy`"

> What about admin, shouldn't there be a default security policy for admin apps as well?

Yes, but this config won't have effect on them, CSP will always be enforced in the admin apps.

**Comment by rymsha** [comment]
> Also, how does this relate to the "default headers" config that can be set for the web config?

It is not related to default headers at all.

> A bit unclear this one: "page.defaultContentSecurityPolicy" - Does it apply to site engine, webapps, admin - or all? If this header only applies to site engine, I think it should be called "site.default.." And explicitly state it only relates to site engine.

it is complicated...
There are two endpoints `/site` and `/admin/site`. 
`/site` endpoint is not affected at all by this configuration. Calling it `site.defaultContentSecurityPolicy` would be confusing.
`/admin/site` - is affected. 
They both relay the work to `PageHandler` - which is the "real" engine. That is why it is called `page.`

None of `/_/` endpoints (components, webapps, services) are affected by this config.

File: docs/release/upgrade.adoc [hunk 1]
```diff
@@ -46,6 +46,9 @@ We generally always recommend updating your apps to the latest version!
 
 == v7.9 notes
 
+* <<deployment/config#_portal, Portal config>> now specifies default Content-Security-Policy for page `preview`/`inline`/`edit` mode (it does not affect `live` mode) .
+It is possible to turn the header off or configure via `page.defaultContentSecurityPolicy` value.
```
Comment (alansemenov) [inline docs/release/upgrade.adoc:50]: It is possible to configure the header via `page.defaultContentSecurityPolicy` value or go back to previous behaviour by setting `page.defaultContentSecurityPolicy` to an empty value.

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: docs/release/upgrade.adoc (hunk 1 above)
Comment (rymsha) [inline docs/release/upgrade.adoc:50]:
```suggestion
It is possible to configure the header via `page.defaultContentSecurityPolicy` value or go back to previous behavior by setting `page.defaultContentSecurityPolicy` to an empty value.
```

**Review by alansemenov — APPROVED** [review]

**Pull request merged**